### PR TITLE
Implemented both the new setup for assessments and added setters to A…

### LINKED ISSUE #324: Give assessment results setters for text and score
After discussing with @omarreiss, it was decided upon to add setters to the AssessmentResult object to ensure cleaner code.

### PR DESCRIPTION
**Pull request by jcomack**
Implemented both the new setup for assessments and added setters to AssessmentResult

Fixes #324

### CODE REVIEW

File: js/values/AssessmentResult.js [hunk 1]
```diff
@@ -43,4 +55,13 @@ AssessmentResult.prototype.getText = function() {
 	return this.text;
 };
 
+/**
+ * Set the text for the assessment.
+ * @param {string} text The text to be used for the text property
+ * @returns {void}
+ */
+AssessmentResult.prototype.setText = function( text ) {
+	this.text = text;
```
Comment (terw-dan) [inline js/values/AssessmentResult.js]: We could include a check for an undefined string, and set it as an empty string, so it matches the default.

**Comment by terw-dan** [comment]
CR Done

File: js/assessments/countWords.js [hunk 2]
```diff
@@ -1,74 +1,82 @@
 var AssessmentResult = require( "../values/AssessmentResult.js" );
 var inRange = require( "lodash/number/inRange" );
 
-var countWordsAssessment = function( paper, researcher, i18n ) {
-	var score = 0;
-	var text = "";
-	var wordCount = researcher.getResearch( "wordCount" );
-
+var calculateWordCountResult = function( wordCount, i18n ) {
```
Comment (andizer) [inline js/assessments/countWords.js]: Missing documentation.

File: js/assessments/countWords.js [hunk 3]
```diff
@@ -1,74 +1,82 @@
...
+				"js-text-analysis",
+				"The text contains %1$d word, this is far too low and should be increased.",
+				"The text contains %1$d words, this is far too low and should be increased.",
+				wordCount
+			)
+		};
 	}
+};
+
+var countWordsAssessment = function( paper, researcher, i18n ) {
```
Comment (andizer) [inline js/assessments/countWords.js]: Documentation?

**Comment by andizer** [comment]
Cr done.

**Pull request merged**